Makefile

```diff
@@ -171,6 +171,8 @@ LDFLAGS		+=	$(platform-ldflags-y)
 LDFLAGS		+=	$(firmware-ldflags-y)
 
 MERGEFLAGS	+=	-r
+MERGEFLAGS	+=	-b elf$(PLATFORM_RISCV_XLEN)-littleriscv
+MERGEFLAGS	+=	-m elf$(PLATFORM_RISCV_XLEN)lriscv
 
```

Comment:
Small correction ...

emulation mode will be "elf$(PLATFORM_RISCV_XLEN)lriscv_$(PLATFORM_RISCV_ABI)"

Reply:
You'll notice that this convention doesn't exist on newlib:
```riscv64-unknown-elf-ld: supported emulations: elf64lriscv elf32lriscv```

Also lp64d/ilp32d are not used, the default elf64lriscv / elf32lriscv implies the d extension, for more check out.
https://github.com/bminor/binutils-gdb/blob/master/ld/emulparams/elf32lriscv.sh
https://github.com/bminor/binutils-gdb/blob/master/ld/emulparams/elf64lriscv.sh
So for example if PLATFORM_RISCV_ABI is lp64d your approach will fail since there is no elf64lriscv_lp64d emulation.
```riscv64-unknown-linux-gnu-ld: supported emulations: elf64lriscv elf64lriscv_lp64f elf64lriscv_lp64 elf32lriscv elf32lriscv_ilp32f elf32lriscv_ilp32```

I think staying with the default one that implies the d extension (since it's needed anyway for OpenSBI) is more portable and cleaner, at least for now. If OpenSBI supports builds without the d extension then we could add more logic there to properly handle emulation.

Reply:
Sure, no problem.

Let's use "elf$(PLATFORM_RISCV_XLEN)lriscv" as emulation mode.